src/creature.cpp

```diff
@@ -207,8 +207,9 @@ bool Creature::sees( const Creature &critter ) const
     if( wanted_range <= 1 &&
         ( posz() == critter.posz() || g->m.valid_move( pos(), critter.pos(), false, true ) ) ) {
         return visible( p );
+    } else if( ( wanted_range > (perception / 3) && critter.has_flag( MF_NIGHT_INVISIBILITY ) && g->m.light_at( critter.pos() ) <= LL_LOW ) ) {
```

Comment:
Does that treat NV towards the perception number?

Reply:
No.

Reply:
No, but see the updated original post. I believe that shady zombies have a special, blob-given ability that passively distorts light in shady areas, rendering the zombie invisible (think of this ability similar to the passive effect of an incandesant husk).

High-perception characters notice the distortion around the shady zombie, which allows them to locate where it is, despite not really seeing it (they would see, or perhaps imagine, a vague outline of the body).